[netsurf] Add netsurf package

alex0809 opened the pull request:
- The toltec postprocessing script does not like broken symlinks. Unfortunately, the build of netsurf seems to produce some, so I am manually deleting them.
- I am installing and compiling the dependencies with each build in the install_dependencies.sh script. You can have a look in there to see what is built. An alternative might be to include them in some version of the build images.

alex0809 committed:
Add netsurf package

alex0809 committed:
Fix formatting

alex0809 committed:
Merge branch 'testing' into package/netsurf

Eeems commented on package/netsurf/netsurf.draft:
```diff
@@ -0,0 +1,7 @@
+name=NetSurf
+desc=a lightweight web browser
+call=/opt/bin/netsurf
+term=killall -9 netsurf
```
This likely isn't needed for anything other than draft. Both Oxide and remux will properly kill applications without use of this.

alex0809 replied:
Should I remove the line entirely then? I can see some other packages leave it empty like recrossable
`term=:`

Eeems replied:
On older versions of draft it had to exist, which is why they do that. Up to you I guess on if you care about supporting draft or not. I know that @dixonary is working on a newer more modern version of draft, but I don't know when that'll be released.

Eeems commented on package/netsurf/package:
```diff
@@ -0,0 +1,64 @@
+)
+
+build() {
+    scripts/install_dependencies.sh
```
@matteodelabre thoughts on this?

matteodelabre replied:
It would be better to take advantage of Toltec’s dependency management capabilities to handle this.

* Use the `makedepends=(build:a build:b …)` syntax to specify Debian packages that need to be installed before compiling. See [the docs](https://github.com/toltec-dev/toltec/blob/testing/docs/package.md#makedepends) for more info on this.
* Add new recipes for libraries on which this package depends (libcurl, libfreetype, libjpeg-turbo—from what I can see). See what was done with [libdlib](https://github.com/toltec-dev/toltec/blob/testing/package/libdlib/package) and [recrossable](https://github.com/toltec-dev/toltec/blob/testing/package/recrossable/package) for example.
* Note that a CMake toolchain file is already included in the Toltec build toolchain, so you don’t need to recreate it. See [vnsee](https://github.com/toltec-dev/toltec/blob/testing/package/vnsee/package) for an example of a project that uses CMake.

alex0809 replied:
OK thanks for the links, @matteodelabre.
Would you prefer to handle the new recipes as separate pull requests each or with this one?

Also, I'll be happy to create the new packages for these dependencies, but I'm not so comfortable with setting myself as maintainer for them because I can't foresee how much free time I have available for this longer-term.

Eeems replied:
I believe you can leave the maintainer blank and one of us will pick it up when we got around to approving it.

raisjn replied:
> Add new recipes for libraries on which this package depends (libcurl, libfreetype, libjpeg-turbo—from what I can see). See what was done with libdlib and recrossable for example.

>  Would you prefer to handle the new recipes as separate pull requests each or with this one?

i think this (making new packages for each dep) should only be done if alex has time + cycle. we should first review the package, then refactor the dependencies in a separate PR, so this PR doesn't get held up.

Eeems replied:
@raisjn For sure, all I'm waiting on is moving the build dependencies to `makedepends`. Which looks like it got committed a little while ago, so now I'm waiting for a build with it to test.

Eeems commented on package/netsurf/package:
```diff
@@ -0,0 +1,64 @@
+    scripts/install_dependencies.sh
+    TARGET_WORKSPACE=$(pwd)/build scripts/build.sh
+
+    # After build, the Netsurf directory contains some broken symlinks, delete them before continuing
```
What were these symlinks for?

alex0809 replied:
Something entirely unrelated, some links in the GTK and RISC OS frontends.

```
./build/netsurf/frontends/gtk/res/Messages
./build/netsurf/frontends/riscos/appdir/Resources/en/Messages
./build/netsurf/frontends/riscos/appdir/Resources/nl/Messages
./build/netsurf/frontends/riscos/appdir/Resources/it/Messages
./build/netsurf/frontends/riscos/appdir/Resources/de/Messages
./build/netsurf/frontends/riscos/appdir/Resources/fr/Messages
```
I haven't touched these frontends at all.

Eeems replied:
Maybe update the comment with that information for future reference?

alex0809 replied:
I just had a closer look and those symlinks are already broken in the git repository. So I'll just remove them from there entirely.

Eeems commented:
Works fine on a rM1

alex0809 committed:
Update to version 0.3.4, remove now unnecessary symlink deletion

Eeems reviewed (dismissed):
Switch over to using the toltec dependency system.

matteodelabre committed:
Move website generation to separate repository (#340)

This PR removes all the logic and resources related to generating the Toltec website. These have been moved to the toltec-dev/web repository.

This change enables us to iterate on the website without being constrained by the stable release cycle from the main repository.

To make sure that the website always contains the latest stable bootstrap script and checksum, the stable build workflow of this repository sends `repository_dispatch` events to the toltec-dev/web repository. See <https://github.community/t/triggering-by-other-repository/16163/2> for details.

alex0809 committed:
Use dependencies instead of install script

alex0809 committed:
Add additional resources, to fix the CSS on internal pages

alex0809 committed:
Change the license to GPL-2

The build scripts in the netsurf-reMarkable repository are licensed
under MIT license, but netsurf itself is GPL-2. Hence I think the
license of the package must be GPL-2.

alex0809 committed:
Update netsurf to version 0.3.5

This new version fixes the display of local files (this is also
important because it finally allows user access of the licenses), and
change the default start page to a new custom about:welcome page.

alex0809 committed:
Fix formatting

alex0809 committed:
Merge branch 'testing' into package/netsurf

alex0809 committed:
Move back to a build without dependencies

This moves back to calling the install_dependencies.sh script for
building and installing the dependencies.
Unlike previous builds, these are then statically linked.

alex0809 commented:
The latest commit moves back to building and installing the dependencies via install_dependencies.sh script, due to some unresolved general open questions about how build dependencies should be handled in toltec.

It's not quite back to square one, because now libcurl, libfreetype and libjpeg-turbo are statically linked, so netsurf does not depend on the library versions available on the reMarkable.

Eeems commented:
> The latest commit moves back to building and installing the dependencies via install_dependencies.sh script, due to some unresolved general open questions about how build dependencies should be handled in toltec.

You haven't fully moved back as you are still using build dependencies for installing packages with apt, which is still how we expect this to work.

alex0809 commented:
> > The latest commit moves back to building and installing the dependencies via install_dependencies.sh script, due to some unresolved general open questions about how build dependencies should be handled in toltec.
> 
> You haven't fully moved back as you are still using build dependencies for installing packages with apt, which is still how we expect this to work.

Yes that's true, I forgot that with the initial commit I installed everything in my script. So it's still a step forward!

alex0809 committed:
Merge branch 'testing' into package/netsurf

alex0809 committed:
Update version to 0.3.8, add missing netsurf logo

alex0809 committed:
Merge branch 'testing' into package/netsurf

alex0809 committed:
Update netsurf package to latest version 0.4

Eeems committed:
Merge branch 'testing' into package/netsurf

raisjn committed:
Merge branch 'testing' into package/netsurf

Eeems committed:
Merge branch 'testing' into package/netsurf
